tests/tests/test-precompile-assets-erc20.ts

```diff
@@ -0,0 +1,410 @@
+};
+
+async function mockAssetBalance(context, assetBalance, assetDetails, sudoAccount, assetId) {
+  // Register the asset
+  await context.polkadotApi.tx.sudo
+    .sudo(
+      context.polkadotApi.tx.assetManager.registerAsset(
+        sourceLocationRelayAssetType,
+        relayAssetMetadata,
+        new BN(1)
+      )
+    )
+    .signAndSend(sudoAccount);
+  await context.createBlock();
+
+  let assets = (
+    (await context.polkadotApi.query.assetManager.assetIdType(assetId)) as any
+  ).toJSON();
+  // make sure we created it
+  expect(assets["xcm"]["parents"]).to.equal(1);
+
+  // Get keys to modify balance
+  let module = xxhashAsU8a(new TextEncoder().encode("Assets"), 128);
+  let account_key = xxhashAsU8a(new TextEncoder().encode("Account"), 128);
+  let blake2concatAssetId = new Uint8Array([
+    ...blake2AsU8a(assetId.toU8a(), 128),
+    ...assetId.toU8a(),
+  ]);
+  let blake2concatAccount = new Uint8Array([
+    ...blake2AsU8a(hexToU8a(ALITH), 128),
+    ...hexToU8a(ALITH),
+  ]);
+  let overallAccountKey = new Uint8Array([
+    ...module,
+    ...account_key,
+    ...blake2concatAssetId,
+    ...blake2concatAccount,
+  ]);
+
+  // Get keys to modify total supply
+  let assetKey = xxhashAsU8a(new TextEncoder().encode("Asset"), 128);
+  let overallAssetKey = new Uint8Array([...module, ...assetKey, ...blake2concatAssetId]);
+
+  await context.polkadotApi.tx.sudo
+    .sudo(
+      context.polkadotApi.tx.system.setStorage([
+        [u8aToHex(overallAccountKey), u8aToHex(assetBalance.toU8a())],
+        [u8aToHex(overallAssetKey), u8aToHex(assetDetails.toU8a())],
+      ])
+    )
+    .signAndSend(sudoAccount);
+  await context.createBlock();
+  return;
+}
+
+const ADDRESS_ERC20 = "0xFfFFfFff1FcaCBd218EDc0EbA20Fc2308C778080";
+const SELECTORS = {
+  balanceOf: "70a08231",
+  totalSupply: "18160ddd",
+  approve: "095ea7b3",
+  allowance: "dd62ed3e",
+  transfer: "a9059cbb",
+  transferFrom: "23b872dd",
+  logApprove: "0x8c5be1e5ebec7d5bd14f71427d1e84f3dd0314c0f7b2291e5b200ac8c7c3b925",
+  logTransfer: "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
+};
+const GAS_PRICE = "0x" + (1_000_000_000).toString(16);
+
+describeDevMoonbeam(
+  "Precompiles - Assets-ERC20 Wasm",
+  (context) => {
+    let sudoAccount, assetId, iFace;
+    before("Setup contract and mock balance", async () => {
+      const keyring = new Keyring({ type: "ethereum" });
+      sudoAccount = await keyring.addFromUri(ALITH_PRIV_KEY, null, "ethereum");
+      // We need to mint units with sudo.setStorage, as we dont have xcm mocker yet
+      // And we need relay tokens for issuing a transaction to be executed in the relay
+      const balance = context.polkadotApi.createType("Balance", 100000000000000);
+      const assetBalance = context.polkadotApi.createType("AssetBalance", { balance: balance });
+
+      assetId = context.polkadotApi.createType(
+        "AssetId",
+        new BN("42259045809535163221576417993425387648")
+      );
```

Comment:
assetId could be deduced by the result of registerAsset.
Otherwise it would be more readable to declare this number as a constant: `const testAssetId=new BN("42259045809535163221576417993425387648")`